ui/src/app/edge/history/thermometer/widget.component.ts

```diff
@@ -0,0 +1,29 @@
+import { ActivatedRoute } from '@angular/router';
+import { Edge, Service } from '../../../shared/shared';
+import { Component, Input, OnInit } from '@angular/core';
+import { DefaultTypes } from 'src/app/shared/service/defaulttypes';
+
+@Component({
+    selector: ThermometerWidgetComponent.SELECTOR,
+    templateUrl: './widget.component.html'
+})
+export class ThermometerWidgetComponent implements OnInit {
+
+    @Input() public period: DefaultTypes.HistoryPeriod;
+
+    private static readonly SELECTOR = "thermometerWidget";
+
+    public edge: Edge = null;
```

Comment:
Union type is recommended for strict null checks https://www.typescriptlang.org/docs/handbook/advanced-types.html#optional-parameters-and-properties